### PR TITLE
Swap module simulation scaffolding

### PR DESCRIPTION
**Pull request by denalimarsh**
Implements [swap sims ticket](https://app.asana.com/0/1156716699555540/1200386098666785/f)

### CODE REVIEW

File: x/swap/simulation/genesis.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	tokens := []string{genTokenDenom(r), genTokenDenom(r)}
+	sort.Strings(tokens)
+	return tokens[0], tokens[1]
+}
+
+func genTokenDenom(r *rand.Rand) string {
+	denom := strings.ToLower(simulation.RandStringOfLength(r, 3))
+	err := sdk.ValidateDenom(denom)
+	for err != nil {
+		denom = strings.ToLower(simulation.RandStringOfLength(r, 3))
```
Comment (karzak) [inline x/swap/simulation/genesis.go:55]: Can we re-write this to only create valid denoms?

File: x/swap/simulation/params.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import (
+	"math/rand"
+
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+)
+
+// ParamChanges defines the parameters that can be modified by param change proposals
+// on the simulation
+func ParamChanges(r *rand.Rand) []simulation.ParamChange {
+	return []simulation.ParamChange{}
```
Comment (karzak) [inline x/swap/simulation/params.go:12]: We should define this function to allow pools and swap fee to be changed. https://github.com/cosmos/cosmos-sdk/blob/v0.39.2/x/mint/simulation/params.go#L22 has good example

File: x/swap/simulation/genesis.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+			tokenA, tokenB = genTokenDenoms(r)
+		}
+		newPool := types.NewAllowedPool(tokenA, tokenB)
+		pools[i] = newPool
+	}
+	return pools
+}
+
+func genTokenDenoms(r *rand.Rand) (string, string) {
+	tokens := []string{genTokenDenom(r), genTokenDenom(r)}
```
Comment (karzak) [inline x/swap/simulation/genesis.go:46]: should check denom[0] != denom[1]

**Review by karzak — CHANGES_REQUESTED** [review]

File: x/swap/simulation/genesis.go (hunk 3 above)
Comment (denalimarsh) [reply x/swap/simulation/genesis.go]: https://github.com/Kava-Labs/kava/pull/924/commits/8621d97b877d15a420b044b82b183a573c3226ae

File: x/swap/simulation/params.go (hunk 2 above)
Comment (denalimarsh) [reply x/swap/simulation/params.go]: https://github.com/Kava-Labs/kava/pull/924/commits/71d0eab53138120edc4ebecbd521af0721266184

**Pull request merged**